From 4f7092ba05cccfeea7d26177d93f3d7a57d9bce2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 8 Jun 2015 14:29:03 +0200
Subject: [PATCH 3/4] cmake: Add an ordering dep between HTML & man Sphinx
 targets

Add a dependency between HTML & manpage Sphinx targets to prevent two
instances of Sphinx from running in parallel, and therefore solves race
conditions reusing the same doctree directory.

Fixes: https://llvm.org/bugs/show_bug.cgi?id=23781
---
 docs/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index da27627..63309a6 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -100,6 +100,9 @@ if (LLVM_ENABLE_SPHINX)
 
     if (${SPHINX_OUTPUT_MAN})
       add_sphinx_target(man llvm)
+      if (${SPHINX_OUTPUT_HTML})
+        add_dependencies(docs-llvm-html docs-llvm-man)
+      endif()
     endif()
 
   endif()
-- 
2.4.3

